tools/flaky-test-reporter/slack_notification.go

```diff
@@ -109,13 +109,23 @@ func createSlackMessageForRepo(rd *RepoData, flakyIssuesMap map[string][]*flakyI
 	flakyTests := getFlakyTests(rd)
 	message := fmt.Sprintf("As of %s, there are %d flaky tests in '%s'",
 		time.Unix(*rd.LastBuildStartTime, 0).String(), len(flakyTests), rd.Config.Repo)
-	for _, testFullName := range flakyTests {
-		message += fmt.Sprintf("\n>- %s", testFullName)
-		if flakyIssues, ok := flakyIssuesMap[getIdentityForTest(testFullName, rd.Config.Repo)]; ok {
+	flakyRate := getFlakyRate(rd)
+	if flakyRate > threshold { // Don't list each test as this can be huge
+		message += fmt.Sprintf("\n>- skip displaying all tests as flaky rate above '%0.2f%%'", threshold)
+		if flakyIssues, ok := flakyIssuesMap[getBulkIssueIdentity(rd, flakyRate)]; ok {
 			for _, fi := range flakyIssues {
```

Comment:
Move the loop to outside the if, then remove the same loop in line 124. Add a comment here expliciting that only one issue is created.

Reply:
Comment added. Moving the loop out might not be as clean as what it currently is, due to another for loop in else statement at line 123: `for _, testFullName := range flakyTests {`

Reply:
Ah, yes.